plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/listener/SpringEventSource.java

```diff
@@ -51,4 +52,49 @@ public void fireEvent(SpringEvent event) {
             }
         }
     }
+
+    /**
+     * check it need skip notify:
+     * 1. if listener is not filterBeanFactory, return false
+     * 2. check it is the same beanFactory for listener and event
+     * @param listener
+     * @param event
+     * @return
+     */
+    private boolean needSkipNotify(SpringListener listener, SpringEvent event) {
```

Comment:
Instead of introducing 'needSkipNotify' and 'checkSameBeanFactory' methods here, I suggest an alternative impl in 'SpringListener':

remove the following code:

```java
default boolean isFilterBeanFactory() {
        return true;
    }
```

and add the following:

```java
    default boolean shouldSkip(SpringEvent<?> event) {
        return isParent(beanFactory(), event.getBeanFactory());
    }

    static boolean isParent(ConfigurableListableBeanFactory beanFactory,
                            ConfigurableListableBeanFactory sourceBeanFactory) {
        if (beanFactory == null) {
            return true;
        }

        if (sourceBeanFactory == beanFactory) {
            return false;
        }

        if (sourceBeanFactory.getParentBeanFactory() instanceof ConfigurableListableBeanFactory) {
            return isParent(beanFactory, (ConfigurableListableBeanFactory) sourceBeanFactory.getParentBeanFactory());
        }

        return true;
    }
```

By doing this, the code is cleaner.